core/shared/src/main/scala/io/circe/Decoder.scala

```diff
@@ -301,6 +301,10 @@ final object Decoder extends TupleDecoders with LowPriorityDecoders {
     c.focus match {
       case JNull => Xor.right(Float.NaN)
       case JNumber(number) => Xor.right(number.toDouble.toFloat)
+      case JString(string) => Xor.fromOption(
+        JsonNumber.fromString(string).map(_.toDouble.toFloat),
```

Comment:
`toDouble.toFloat` may overflow and result in `Infinity`, e.g. on `Double.MaxValue = 1.7976931348623157E308`. Since the other number conversions now fail instead of truncating/overflowing, I think this one should too.